src/LibraryManager/Providers/FileSystem/FileSystemProvider.cs

```diff
@@ -100,14 +110,56 @@ public ILibraryCatalog GetCatalog()
         }
 
         /// <summary>
-        /// No-op for FileSystemProvider
+        /// Updates file set on the passed in ILibraryInstallationState in case user selected to have all files included
         /// </summary>
         /// <param name="desiredState"></param>
         /// <param name="cancellationToken"></param>
         /// <returns></returns>
-        public Task<ILibraryInstallationResult> UpdateStateAsync(ILibraryInstallationState desiredState, CancellationToken cancellationToken)
+        public async Task<ILibraryInstallationResult> UpdateStateAsync(ILibraryInstallationState desiredState, CancellationToken cancellationToken)
         {
-            return Task.FromResult<ILibraryInstallationResult>(LibraryInstallationResult.FromSuccess(desiredState));
+            if (cancellationToken.IsCancellationRequested)
+            {
+                return LibraryInstallationResult.FromCancelled(desiredState);
+            }
+
+            try
+            {
+                var catalog = (FileSystemCatalog)GetCatalog();
+                ILibrary library = await catalog.GetLibraryAsync(desiredState.LibraryId, cancellationToken).ConfigureAwait(false);
+
+                if (library == null)
+                {
+                    throw new InvalidLibraryException(desiredState.LibraryId, Id);
+                }
+
+                if (desiredState.Files != null && desiredState.Files.Count > 0)
```

Comment:
What if desiredState.LibraryId is a file path and desiredState.Files contains different file paths. This currently implementation will use the desiredState.Files instead of desiredState.LibraryId. We should error if desiredState.LibraryId is a file path and Files is specified and is different from desiredState.LibraryId.

Reply:
I would think Files should be relative paths under LibraryId if LibraryId is a directory. If LibraryId is a file, then Files should be empty.

Reply:
Jimmy told me that the original implementation had a design to copy the LibraryId file contents to the destination with the Files: property names. Basically, a way to rename files. Not sure this is a clean design and that the outcome of this operation is what the user expects. E.g.: Library 
{
"library": "path/to/file.txt"
"destination": "folder1",
"provider": "filesystem",
"files" : ["newFileName.txt"]
}

This would lead to a new file being created at .\folder1\newFileName.txt with the path/to/file.txt contents. This is what the current implementation does and I could leave as it for now.